packages/astro/src/@types/astro.ts

```diff
@@ -1375,6 +1375,91 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		devOverlay?: boolean;
+
+		// TODO review with docs team before merging to `main`
+		/**
+		 * @docs
+		 * @name experimental.i18n
+		 * @type {object}
+		 * @version 3.5.0
+		 * @type {object}
+		 * @description
+		 *
+		 * Configures experimental i18n routing and allows you to specify some customization options.
+		 */
+		i18n?: {
+			/**
+			 * @docs
+			 * @name experimental.i18n.defaultLocale
+			 * @type {string}
+			 * @version 3.5.0
+			 * @description
+			 *
+			 * The default locale of your website/application
+			 */
+			defaultLocale: string;
+			/**
+			 * @docs
+			 * @name experimental.i18n.locales
+			 * @type {string[]}
+			 * @version 3.5.0
+			 * @description
+			 *
+			 * A list of all locales supported by the website. e.g. `['en', 'fr', 'es']` This list should/should not include the `defaultLocale`.
+			 */
+			locales: string[];
+
+			/**
+			 * @docs
+			 * @name experimental.i18n.fallback
+			 * @type {Record<string, string>}
+			 * @version 3.5.0
+			 * @description
+			 *
+			 * The content fallback strategy when navigating to pages in non-default languages that do not exist. (e.g. a translated page has not been created).
+			 *
+			 * This a plain object, where a key is a locale that hasn't traslated pages, and the corresponding value is the locale where you want to redirect the missing page to.
+			 *
+			 * #### Examples
+			 *
+			 * The following example configures your content fallback strategy to redirect unavailable pages in `/pt/` to their `es` version, and unavailable pages in `/fr/` to their `en` version. Unavailable `/es/` pages will return a 404.
+			 *
+			 * ```js
+			 * export defualt defineConfig({
+			 * 	experimental: {
+			 * 		i18n: {
+			 * 			defuaultLocale: "en",
+			 * 			locales: ["en", "fr", "pt", "es"],
+			 * 			fallback: {
+			 * 				pt: "es",
+			 * 			  fr: "en"
+			 * 			}
+			 * 		}
+			 * 	}
+			 * })
+			 * ```
+			 */
+			fallback?: Record<string, string>;
+
+			/**
+			 * @docs
+			 * @name experimental.i18n.routingStrategy
+			 * @type {'prefix-always' | 'prefix-other-locales'}
+			 * @default 'prefix-other-locales'
+			 * @version 3.5.0
+			 * @description
+			 *
+			 * Controls the routing strategy to determine your site URLs.
+			 *
+			 *  - `prefix-other-locales`(default): Only non-default languages will display a language prefix. The `defaultLocale` will not show a language prefix.
+			 *    URLs will be of the form `example.com/[lang]/content/` for all non-default languages, but `example.com/content/` for the default locale.
+			 *  - `prefix-always`: All URLs will display a language prefix.
+			 *    URLs will be of the form `example.com/[lang]/content/` for every route, including the default language.
+			 *
+			 * Note: Astro requires all content to exist within a `/[lang]/` folder, even for the default language.
+			 */
+			routingStrategy: 'prefix-always' | 'prefix-other-locales';
+		};
```

Comment:
Thank you @delucis , and don't worry if you missed the discussion/context. 

That's excellent feedback, thank you. The reason why I used "prefix" is because it seems to be the most common term used in this context. Personally, I don't like it because it introduces a new terminology that isn't very clear in the space of "routing and internationalization".

I will have more context and information once domain support is clearer. 

I would like to avoid breaking down the feature in **too many** flags because they are a footgun, and they become more difficult to maintain. They essentially become a matrix. In fact, `base`, `trailingSlash` and `build.format` are very difficult to maintain together.

I can see have at most **two** properties,  as you suggested, but like this:

```ts
interface i18n {
	routingStrategy: {
		subUrl?: 'suburl-all' | 'suburl-non-default',
		subDomain?: 'subdomain-all' | 'subdomain-non-default'
	}
}
```
I used "URL" because that's what the users see, and "path" is misleading because it can be used for files too.

This could allow us to have support cases like this:
- `en.example.com/`
- `en.example.com/en-au`

FYI, if we agree on something,  **I don't** plan to apply this change in this PR. I would like to apply these changes **after** we ship the experimental release. 

What do you think?

Reply:
I’ll move this discussion to the RFC too 👍